xmtp_mls/src/groups/subscriptions.rs

```diff
@@ -121,21 +118,21 @@ impl MlsGroup {
         message.ok_or(GroupError::MissingMessage)
     }
 
-    pub async fn stream<ApiClient>(
-        &self,
-        client: Arc<Client<ApiClient>>,
-    ) -> Result<Pin<Box<dyn Stream<Item = StoredGroupMessage> + Send + '_>>, GroupError>
+    pub async fn stream<'a, ApiClient>(
+        &'a self,
+        client: &'a Client<ApiClient>,
+    ) -> Result<impl Stream<Item = StoredGroupMessage> + '_, GroupError>
```

Comment:
This lifetime capture (`+ '_`) becomes the default behavior in rust 2024 edition + allows us to attach more flexible lifetime captures wherever we use `impl Trait`, which will allow eliding the `'a`lifetime here